CoffeeMachineSimulator/CoffeeMachineSimulator.Tests/Services/CoffeeServiceTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using CoffeeMachineSimulator.Services.Models;
+using CoffeeMachineSimulator.Services.Services;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CoffeeMachineSimulator.Tests.Services
+{
+    [TestFixture]
+    public class CoffeeServiceTests
+    {
+        private CoffeeService coffeeService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            coffeeService = new CoffeeService();
+        }
+
+        [Test]
+        public void GetCoffees_Returns_ListOfCoffeeModels()
+        {
+            var coffeesReturned = coffeeService.GetCoffees();
+
+            Assert.IsNotNull(coffeesReturned);
+            Assert.IsTrue(coffeesReturned.Any());
+            Assert.IsInstanceOf(typeof(List<CoffeeModel>), coffeesReturned);
+            Assert.IsTrue(coffeesReturned.Any(coffeeModel => coffeeModel.Id != Guid.Empty && !string.IsNullOrEmpty(coffeeModel.Name)));
+        }
+
+        [Test]
+        public void AddNullCoffee_Throws_NewException()
+        {
+            var ex = Assert.Throws<Exception>(() => coffeeService.AddCoffee(null));
+            
+            Assert.AreEqual(ex.Message, "You should not add null entries!");
+        }
+
+        [Test]
+
+        //TODO: Add Unit test for Remaining add cases
+        public void AddExistingCoffee_Throws_NewException()
+        {            
+            var coffeeadded = new CoffeeModel(); coffeeadded = coffeeService.GetCoffees()[0];
+            var ex= Assert.Throws<Exception>(() => coffeeService.AddCoffee(coffeeadded));
+
+            Assert.AreEqual(ex.Message, "Already exists an entry with the same Id");
+        }
+
+        [Test]
+        public void AddEmptyFieldCoffee_Throws_NewException()
```

Comment:
Here I would create a new coffeeModel on which I would set all of these values that you want to check. Later I would send that object to the AddCoffee() method. For the checks I would create a new method which would return a boolean of all of the properties that you set, are in the correct format and expected result (ex: bool IsCoffeModelValid(CoffeeModel coffee) ). Then I would assert that, that method returns true for a specific object that you created previously.
You can leave it like this also, it's just a matter of structuring the code. 
One thing that should definitely change is the instantiation of the models. It's enough to have only one coffee model and work with that.